kvbc/tools/db_editor/include/kv_blockchain_db_editor.hpp

```diff
@@ -392,6 +392,40 @@ struct GetCategoryEarliestStale {
   }
 };
 
+struct GetStaleKeysSummary {
+  const bool read_only = true;
+  std::string description() const {
+    return "getStaleKeysSummary [BLOCK-VERSION-TO]\n"
+           "  return the number of stale keys per category type in the current blockchain\n"
+           "If BLOCK-VERSION-TO has been given, the method will summerize the stale keys from current genesis to "
+           "BLOCK-VERSION-TO";
+  }
+
+  std::string execute(KeyValueBlockchain &adapter, const CommandArguments &args) const {
+    auto latestBlockID = adapter.getLastReachableBlockId();
+    if (args.values.size() >= 1) {
+      latestBlockID = toBlockId(args.values.front());
+    }
+    const auto &categories = adapter.blockchainCategories();
+    std::map<CATEGORY_TYPE, uint64_t> stale_keys_per_category_type_;
+    for (const auto &[cat_id, cat_type] : categories) {
+      (void)cat_id;
+      stale_keys_per_category_type_.emplace(cat_type, 0);
+    }
+    for (auto block = adapter.getGenesisBlockId(); block <= latestBlockID; block++) {
+      auto stale_keys = adapter.getBlockStaleKeys(block);
+      for (const auto &[cat_id, cat_type] : categories) {
+        stale_keys_per_category_type_[cat_type] += stale_keys[cat_id].size();
+      }
+    }
+    std::map<std::string, std::string> out;
+    for (auto const &[cat_type, num_of_stale] : stale_keys_per_category_type_) {
+      out[cat_type_str.at(cat_type)] = std::to_string(num_of_stale);
```

Comment:
Now that I think of it, since we can have multiple categories of the same type, we should return a map of `category ID` -> `count` instead.

Reply:
The requirement is to have the number of stale key per category type and not per category ID. This way, the testing team can compare the number of stale keys in, for example, merkle tree keys and the number of deleted of that type

Reply:
I understand, but, in general, I think it makes sense to expose this information to users. Currently, there is a use case where we have one category of every type, but that might change.

We can leave as is for now, but I think it won't hurt to expose every separate category and its type. Users can then just sum.